docs/blog/2019-03-27-security-concerns-unique-to-static-sites/index.md

```diff
@@ -0,0 +1,128 @@
+
+What does this API allow? First, you can secure your database. Any API keys can be secured on your serverless...server 😭. The point is, they will be secure. But what does this gain you? After all, instead of pillaging your database directly you have given them an API through which to pillage.
+
+Control, you have gained the ability to add rate limiting, better logging, auto blocking of IPs, etc. You can throw the proverbial "secure API handbook" (except for the authentication part) at your users to make sure they behave. More aggressively, you could use a secure cookie for your site to uniquely and anonymously identify each user. This would assist in the previously mentioned security practices and, while not hard to delete a cookie, adds another layer of security.
+
+In the end if you need hardened security, you need authentication. If you want a simpler solution for simpler features, these suggestions might just help. Remember, if you need access a database and you already have authentication, do not ship those secure API keys to the client!
+
+### Exceptions
+
+Every good rule has its exceptions. Never shipping API keys in client-side code is no exception. Let me be clear, you should never ship **SECURE** API keys (often called secrets) to a client. But, there is such a thing as non-secure/public API keys. These keys are generally used for identification of and not for access control. Firebase authentication is a great example of this.
+
+Your site needs to be able to access your unique firebase instance, to authenticate users. To do this, Firebase uses an API key to identify your app. This key is [designed](https://stackoverflow.com/questions/37482366/is-it-safe-to-expose-firebase-apikey-to-the-public) to be public and their documentation tells you to add it to your client-side code. Firebase has other [controls](https://stackoverflow.com/questions/35418143/how-to-restrict-firebase-data-modification) that determine what an anonymous user is allowed to do with this key.
+
+This is also the case with Cloudinary. While they have a private API key (a secret) required to access private content, they also require an API key for [anonymous uploads](https://cloudinary.com/documentation/upload_images#unsigned_upload). This key is designed to be public and is used as a unique identifier of your cloud instance and not as a security measure.
+
+It can difficult to know whether an API key should be kept secret. If the page giving you the keys does not specify, check the docs. If all else fails use a search engine, someone on Stack Overflow has already asked.
+
+## Conclusion
+
+Remember, keep it **secret**, keep it **safe**! Do not store API keys in your repository and do not ship them in you client-side code. Now go make awesome static sites that are completely secure!
```

Comment:
```suggestion
Remember, keep it **secret**, keep it **safe**! Do not store API keys in your repository and do not ship them in your client-side code. 

Now go make awesome Gatsby sites that are completely secure!
```

Reply:
This seems like a great opportunity to mention the OWASP Top 10 and where people can go to learn more about web application security: https://www.owasp.org/images/7/72/OWASP_Top_10-2017_%28en%29.pdf.pdf

Reply:
@marcysutton Great idea!